bundler/lib/bundler/dsl.rb

```diff
@@ -102,38 +102,43 @@ def gem(name, *args)
 
       # if there's already a dependency with this name we try to prefer one
       if current = @dependencies.find {|d| d.name == dep.name }
-        deleted_dep = @dependencies.delete(current) if current.type == :development
+        if current.requirement != dep.requirement
+          current_requirement_open = current.requirements_list.include?(">= 0")
 
-        unless deleted_dep
-          if current.requirement != dep.requirement
-            return if dep.type == :development
+          if current.type == :development
+            @dependencies.delete(current)
 
+            unless current_requirement_open || dep.type == :development
+              Bundler.ui.warn "A gemspec development dependency (#{dep.name}, #{current.requirement}) is being overriden by a Gemfile dependency (#{dep.name}, #{dep.requirement}).\n" \
+                              "Please remove either of them, or make sure they both have the same requirement\n" \
```

Comment:
Would it be wise to announce this will change later?

Reply:
I updated the message, what do you think?

Reply:
looks better, :muscle: